packages/rum/src/domain/segmentCollection/segment.ts

```diff
@@ -83,15 +94,36 @@ export class Segment {
     replayStats.addRecord(this.metadata.view.id)
     this.metadata.has_full_snapshot ||= record.type === RecordType.FullSnapshot
     sendToExtension('record', { record, segment: this.metadata })
-    this.worker.postMessage({ data: `,${JSON.stringify(record)}`, id: this.id, action: 'write' })
+    this.write(`,${JSON.stringify(record)}`)
   }
 
   flush(reason: FlushReason) {
+    this.write(`],${JSON.stringify(this.metadata).slice(1)}\n`)
     this.worker.postMessage({
-      data: `],${JSON.stringify(this.metadata).slice(1)}\n`,
-      id: this.id,
-      action: 'flush',
+      action: 'reset',
+      streamId: STREAM_ID,
     })
     this.flushReason = reason
   }
+
+  private write(data: string) {
+    this.pendingWriteCount += 1
+    this.worker.postMessage({
+      data,
+      id: this.id,
+      streamId: STREAM_ID,
+      action: 'write',
+    })
+  }
+}
+
+function concatBuffers(buffers: Uint8Array[]) {
```

Comment:
Would it be simpler if the utility code was exposed from core?

Reply:
Let's try it. I put it in core, so now the worker depends on core. I was hesitating about it, because in the future core might very well depend on the worker. We'll see how it goes!